### PR TITLE
Add accepted deposit tokens list to contract.

### PR DESCRIPTION
**Pull request by willjgriff**
Added `acceptedDepositTokens` array which restricts the tokens that can be deposited in exchange for the requestable token. 

There is a max number of accepted tokens which is currently 100 as array iteration is required. Some benchmark tests should be written to determine a reasonable maximum that allows execution of at least `removeToken()` within the block limit to prevent disabling the contract. I expect a max of 100 is on the lower end of what is within the block limit. We may wish to make the maximum modifiable.

An alternative approach that doesn't limit the number of accepted tokens is possible, although it would be at the expense of blockchain bloat (which would likely make the contract more expensive once storage rent is introduced).

A max of 100 has also been added to the number of Token Requests an individual address can have, to prevent potential disabling of the `refundTokenRequest()` and `withdrawTokenRequest()` for individual users.

Forked from the branch in this PR: https://github.com/1Hive/token-request-app/pull/34 and reverted changes not relevant to updating the contract as specified above.

### CODE REVIEW

File: contracts/TokenRequest.sol [hunk 1]
```diff
@@ -66,6 +87,31 @@ contract TokenRequest is AragonApp {
...
+    /**
+    * @notice Add `_token.symbol(): string` to the accepted deposit token request tokens
+    * @param _token token address
+    */
+    function addToken(address _token) external auth(MODIFY_TOKENS_ROLE) {
+        require(isContract(_token), ERROR_ADDRESS_NOT_CONTRACT);
+        require(!acceptedDepositTokens.contains(_token), ERROR_TOKEN_ALREADY_ACCEPTED);
+
+        acceptedDepositTokens.push(_token);
+        require(acceptedDepositTokens.length <= MAX_ACCEPTED_DEPOSIT_TOKENS, ERROR_TOO_MANY_ACCEPTED_TOKENS);
```
Comment (rperez89) [inline contracts/TokenRequest.sol:102]: this require should be avobe the acceptedDepositTokens.push right?

Comment (willjgriff) [reply contracts/TokenRequest.sol:102]: We could put it above and change the `<=` to a `<`. If we put it above it as it is the max size of `acceptedDespositTokens` would be 101 (`MAX_ACCEPTED_DEPOSIT_TOKENS` + 1).

**Comment by lkngtn** [comment]
100 accepted tokens seems like it would be more than enough in practice for any use case I can really imagine. I think in most cases people will only use one or two.

File: contracts/TokenRequest.sol (hunk 1 above)
Comment (rperez89) [reply contracts/TokenRequest.sol:102]: ok is the same, feel free to merge it

**Pull request merged**